src/modules/content-tree/content.tree.module.js

```diff
@@ -45,10 +65,119 @@ export default class ContentTreeModule extends Component {
 
         item.subitems = [...item.subitems, ...location.subitems];
 
+        this.updateSubtreeAfterLoadMoreItems(path);
         successCallback();
         this.forceUpdate();
     }
 
+    updateSubtreeAfterLoadMoreItems(path) {
+        const item = this.findItem(this.items, path.split(','));
+
+        this.updateItemInSubtree(this.subtree[0], item, path.split(','));
+        this.saveSubtree();
+    }
+
+    updateSubtreeAfterItemToggle(path, isExpanded) {
+        const item = this.findItem(this.items, path.split(','));
+
+        if (isExpanded) {
+            this.addItemToSubtree(this.subtree[0], item, path.split(','));
+        } else {
+            this.removeItemFromSubtree(this.subtree[0], item, path.split(','));
+        }
+
+        this.saveSubtree();
+    }
+
+    addItemToSubtree(subtree, item, path) {
+        const parentSubtree = this.findParentSubtree(subtree, item, path);
+
+        if (!parentSubtree) {
+            return;
+        }
+
+        const { subitemsLoadLimit } = this.props;
+
+        parentSubtree.children.push({
+            '_media-type': 'application/vnd.ez.api.ContentTreeLoadSubtreeRequestNode',
+            locationId: item.locationId,
+            limit: Math.ceil(item.subitems.length / subitemsLoadLimit) * subitemsLoadLimit,
+            offset: 0,
+            children: [],
+        });
+    }
+
+    removeItemFromSubtree(subtree, item, path) {
+        const parentSubtree = this.findParentSubtree(subtree, item, path);
+
+        if (!parentSubtree) {
+            return;
+        }
+
+        const index = parentSubtree.children.findIndex((element) => parseInt(element.locationId, 10) === parseInt(path[1], 10));
```

Comment:
The other `element.locationId` doesn't have `parseInt`

Reply:
Yup, it can be removed.